posthog/hogql/printer.py

```diff
@@ -1064,6 +1069,31 @@ def visit_call(self, node: ast.Call):
         if optimized_property_group_call := self.__get_optimized_property_group_call(node):
             return optimized_property_group_call
 
+        # Validate parametric arguments
+        if func_meta := (
+            find_hogql_aggregation(node.name)
+            or find_hogql_function(node.name)
+            or find_hogql_posthog_function(node.name)
+        ):
+            if func_meta.parametric_first_arg:
+                if not node.args:
+                    raise QueryError(f"Missing arguments in function '{node.name}'")
+                # Check that the first argument is a constant string
+                first_arg = node.args[0]
+                if not isinstance(first_arg, ast.Constant):
```

Comment:
You'll wanna do `self.visit(node.args[0])` before checking whether it's an `ast.Constant` I think